feat: UTC-210: Gating predictions validation raising errors behind fflag

mcanu opened the pull request:
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gates most of the features related to predictions validations introduced in https://github.com/HumanSignal/label-studio/pull/8062 behind `fflag_feat_utc_210_prediction_validation_15082025` fflag.

mcanu committed:
Added feature flag to gate raising validation

mcanu committed:
Fix test

jombooth commented on label_studio/data_import/functions.py:
```diff
@@ -73,10 +74,13 @@ def async_import_background(
-            project_import.status = ProjectImport.Status.FAILED
-            project_import.save()
-            return
+            if flag_set('fflag_feat_utc_210_prediction_validation_15082025', user=user):
+                project_import.error = error_message
+                project_import.status = ProjectImport.Status.FAILED
+                project_import.save()
```
looks like this change predates the PR, but I think we should use `update_fields` here for sure

jombooth commented on label_studio/data_import/api.py:
```diff
@@ -519,10 +523,17 @@ def _create_legacy(self, project):
-                    f'Prediction {i}: Invalid task ID {item.get("task")} - task not found in project'
-                )
-                continue
+                if flag_set('fflag_feat_utc_210_prediction_validation_15082025', user=self.request.user):
```
Don't we want `org.created_by` here like our usual strategy for flag_set?

jombooth replied:
(here and other call locations)

mcanu replied:
I'm pretty sure there is no need to use the organization created_by. We set the organization attribute [here](https://github.com/HumanSignal/label-studio/blob/292201317f9d9be09af1548cd3aba090d552f6c5/label_studio/core/feature_flags/utils.py#L8) based on that, and then we can target accordingly in Launch Darkly.
I see it used like this in several other places too, maybe I'm missing something

jombooth replied:
ah, got it. The place where it's most important to pass organization.created_by is in async code where we don't have a request. To be consistent here, assuming this code only runs sync, can we use `user='auto'`?

mcanu replied:
Sorry I don't think I'm following. If we use `user='auto'` here we will lose the per org targeting functionality. This particular line is run sync so there is no problem.

I did have to use the organization.created_by user in the [ImportStorage.add_task](https://github.com/HumanSignal/label-studio/blob/2d0e90678f1c1dc0a146f64a3e087e06cbda6f26/label_studio/io_storages/base_models.py#L503) function.

mcanu replied:
oh, my bad, I didn't know that auto did [this](https://github.com/HumanSignal/label-studio/blob/292201317f9d9be09af1548cd3aba090d552f6c5/label_studio/core/feature_flags/base.py#L94-L98). I'll change it.

jombooth replied:
I don't think that's true - if we pass `auto` we'll [pull the user off the request like the current implementation does](https://github.com/HumanSignal/label-studio/blob/d91109a07127545651339f7707644efbd546d562/label_studio/core/feature_flags/base.py#L94-L98). Basically I just think we have too many patterns for this and would like to standardize on just user auto for sync contexts, and created_by for async contexts

mcanu replied:
Updated!

mcanu committed:
Added update_fields to project import status error save

mcanu committed:
using user=auto in flag_set to simplofy

jombooth commented on label_studio/data_import/functions.py:
```diff
@@ -48,7 +48,8 @@ def async_import_background(
     if project_import.preannotated_from_fields:
         # turn flat task JSONs {"column1": value, "column2": value} into {"data": {"column1"..}, "predictions": [{..."column2"}]
-        tasks = reformat_predictions(tasks, project_import.preannotated_from_fields, project)
+        raise_errors = flag_set('fflag_feat_utc_210_prediction_validation_15082025', user=user)
```
please don't hate me but I think this one needs to be
```suggestion
        raise_errors = flag_set('fflag_feat_utc_210_prediction_validation_15082025', user=project.organization.created_by)
```

jombooth replied:
(here and below)

mcanu replied:
no worries! I'll change it

mcanu committed:
Update label_studio/data_import/functions.py

Co-authored-by: Jo Booth <jo.m.booth@gmail.com>

mcanu committed:
Using project.organization.created_by

jombooth reviewed (approved):
🙏

mcanu commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17000078841)

mcanu committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17000078841

mcanu committed:
Merge branch 'develop' into 'fb-utc-210-fflag-validation'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17000078841
